src/Agent.Listener/SelfUpdater.cs

```diff
@@ -412,6 +412,21 @@ private async Task DownloadLatestAgent(CancellationToken token)
                 }
             }
 
+            if (!String.IsNullOrEmpty(AgentKnobs.DisableAuthenticodeValidation.GetValue(HostContext).AsString()))
+            {
+                Trace.Warning("Authenticode validation skipped for downloaded agent package since it is disabled currently by agent settings.");
+            }
+
+            var isValid = this.VerifyAgentAuthenticode(latestAgentDirectory);
+            if (!isValid)
+            {
+                throw new Exception("Authenticode validation of agent assemblies failed.");
+            }
+            else
+            {
+                Trace.Info("Authenticode validation of agent assemblies passed successfully.");
+            }
```

Comment:
Looks like it should be in `else` block above

Reply:
Fixed